source/code/plugin/in_kube_nodes.rb

```diff
@@ -84,6 +84,13 @@ def enumerate
               record["Labels"] = [items["metadata"]["labels"]]
               record["Status"] = ""
 
+              if !items["spec"]["providerID"].nil? && !items["spec"]["providerID"].empty?
+                record["ProviderID"] = items["spec"]["providerID"]
+              else
+                record["ProviderID"] = "onprem"
```

Comment:
anything in 'record' goes to customer workspace (not telemetry). Is that the intention? If so, did you also add the 'field' 'ProviderID' to KubeNode type ?